packages/medusa/src/repositories/product-category.ts

```diff
@@ -160,4 +158,40 @@ export const ProductCategoryRepository = dataSource
     },
   })
 
-export default ProductCategoryRepository
\ No newline at end of file
+export default ProductCategoryRepository
+
+const scopeChildren = (
+  category: ProductCategory,
+  treeScope: QuerySelector<ProductCategory> = {}
+): ProductCategory => {
+  if (isEmpty(treeScope)) {
+    return category
+  }
+
+  category.category_children = category.category_children.filter(
+    (categoryChild) => {
+      return !Object.entries(treeScope).some(
+        ([attribute, value]) => categoryChild[attribute] !== value
+      )
+    }
+  )
+
+  return category
+}
+
+const sortChildren = (
+  category: ProductCategory,
+  treeScope: QuerySelector<ProductCategory> = {}
+): ProductCategory => {
+  if (category.category_children) {
+    category.category_children = category?.category_children
+      .map(
+        // Before we sort the children, we need scope the children
+        // to conform to treeScope conditions
+        (child) => sortChildren(scopeChildren(child, treeScope), treeScope)
+      )
+      .sort((a, b) => a.rank - b.rank)
+  }
```

Comment:
question: What spawned the need for these changes? Don't see how they relate to the scope of the PR, but I might be missing something

Reply:
Got it - thanks for explaining! 🙏